conda/cli/main_list.py

```diff
@@ -145,6 +145,12 @@ def configure_parser(sub_parsers: _SubParsersAction, **kwargs) -> ArgumentParser
         help="In explicit mode, leave authentication details in package URLs. "
         "They are removed by default otherwise.",
     )
+    p.add_argument(
+        "--check",
+        action="store_true",
+        default=False,
+        help="Exit with code 1 if the queried package(s) cannot be found in the environment.",
+    )
```

Comment:
I understand this is to avoid script breakage but I really dislike this flag and would rather we just set the exitcode correctly for `conda list regex`